perf/_utils.py

```diff
@@ -365,3 +365,32 @@ def pipe_cloexec():
     pipe_cloexec = os.pipe
 
     # In Python 3.4, file descriptors are non-inheritable by default (PEP 446)
+
+
+if MS_WINDOWS:
+    import msvcrt
+    if sys.version_info < (3, 0):
+        import _subprocess
+
+    class HandleOfPipe:
```

Comment:
This object looks very complex just to set a flag on a handle, I don't think that it's needed.